examples/event-listener.php

```diff
@@ -0,0 +1,61 @@
+<?php
+
+namespace event_listener;
+
+use Castor\Attribute\AsListener;
+use Castor\Attribute\AsTask;
+use Castor\Event\BeforeExecuteTaskEvent;
+
+use function Castor\dispatch_event;
+use function Castor\io;
+
+#[AsTask(description: 'Task that need ensure the environment is set')]
+function my_task(): void
+{
+    io()->writeln('Hello from task!');
+}
+
+// Listen to a provided event
+#[AsListener(event: BeforeExecuteTaskEvent::class, priority: 1)]
+function my_listener(BeforeExecuteTaskEvent $event): void
+{
+    $taskName = $event->getCommand()->getName();
+
+    if ('event-listener:my-task' === $taskName) {
+        io()->writeln('Hello from listener! (lower priority)');
+    }
+}
+
+#[AsListener(event: BeforeExecuteTaskEvent::class, priority: 10)]
+function my_listener_that_has_higher_priority(BeforeExecuteTaskEvent $event): void
+{
+    $taskName = $event->getCommand()->getName();
+
+    if ('event-listener:my-task' === $taskName) {
+        io()->writeln('Hello from listener! (higher priority)');
+    }
+}
+
```

Comment:
Would be great to add all the remaining events in the testing. That would also help making sure they're properly called in the right order. (AppInit first, etc.)